legit.py

```diff
@@ -1,54 +1,87 @@
 from git import Repo
 from git.exc import GitCommandError
 from typing import Iterable
+import logging
 
+# ==============================================================
+# create logger with 'legit'
+log = logging.getLogger("legit")
+log.setLevel(logging.DEBUG)
+# create file handler which logs even debug messages
+fh = logging.FileHandler("legit.log")
+fh.setLevel(logging.DEBUG)
+# create console handler which logs info messages
+ch = logging.StreamHandler()
+ch.setLevel(logging.ERROR)
+# create formatter and add it to the handlers
+formatter = logging.Formatter(
+    "%(asctime)s\t%(levelname)s -- %(processName)s %(filename)s:%(lineno)s -- %(message)s"
+)
+ch.setFormatter(formatter)
+fh.setFormatter(formatter)
+# add the handlers to the logger
```

Comment:
Code speaks better what he do here. So it looks that we do not need this comment here.